### ISSUE TITLE
[BUG] Session is blocked in Russia

### ISSUE THREAD

**Issue by keybreak** [session-foundation/session-android#30]
### Code of conduct

- [X] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-android/blob/master/CODE_OF_CONDUCT.md)

### Self-training on how to write a bug report

- [X] I have learned [how to write a bug report](https://www.browserstack.com/guide/how-to-write-a-bug-report)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a friend living in Russia, he's not very tech (GitHub) savvy, so i'd have to be a mediator for him.

Since yesterday both Android and Linux Session can not connect to network, path is always red and rotating.
And, unfortunately, looks like it's censorship ministry of Russia started blocking Session, just like recently they did with Signal (btw it's censorship circumvention toggle seems to work against it), because when running through VPN / Orbot on Android he says it can connect.

That is a big problem, since Session / lokinet network suppose to circumvent censorship and bring freedom against this kind of tyranny.

We need some sort of working circumvention mechanism ASAP, i suggest looking into something like Tor's webtunnel, as it seems to work perfectly so far inside China, Russia and Iran against Tor blockade.

### Session Version

1.19.1 (3795)

**Comment by cn-taoye** [comment]
Blocked in china too

**Comment by KeeJef** [comment]
> Blocked in china too

How exactly is Session being blocked for you in China?

Is it just the seed nodes being blocked? If that’s the case, the app behavior you would be seeing is that Session would fail to establish a path and fail to send or receive messages after its initial download. However, using a VPN briefly would allow Session to retrieve the Service Node list, after which you could turn the VPN off and continue using Session normally.

Or are connections to the Service Nodes themselves being blocked? If so, Session wouldn’t work at all without a VPN.

**Comment by cn-taoye** [comment]
> > Blocked in china too
> 
> How exactly is Session being blocked for you in China?
> 
> Is it just the seed nodes being blocked? If that’s the case, the app behavior you would be seeing is that Session would fail to establish a path and fail to send or receive messages after its initial download. However, using a VPN briefly would allow Session to retrieve the Service Node list, after which you could turn the VPN off and continue using Session normally.
> 
> Or are connections to the Service Nodes themselves being blocked? If so, Session wouldn’t work at all without a VPN.

Yes, previously I could obtain the service node list through a temporary VPN connection, and the session would work normally. However, recently, after disconnecting, the session no longer works.

**Comment by cn-taoye** [comment]
> > Blocked in china too
> 
> How exactly is Session being blocked for you in China?
> 
> Is it just the seed nodes being blocked? If that’s the case, the app behavior you would be seeing is that Session would fail to establish a path and fail to send or receive messages after its initial download. However, using a VPN briefly would allow Session to retrieve the Service Node list, after which you could turn the VPN off and continue using Session normally.
> 
> Or are connections to the Service Nodes themselves being blocked? If so, Session wouldn’t work at all without a VPN.

Yes, previously I could obtain the Service Node List through a temporary VPN connection, and continue using Session normally. But recently, after disconnecting, Session wouldn’t work.

**Comment by PRESFIL** [comment]
Any update on this? Adding socks proxy #130 would be easy solution for the first time.
I know, that VPN works, by ability to specify proxy is more flexible way.

Would be nice to see this feature on all platforms, since it is a cross-platform application 😸
